chore(mimir): update to 3.3.*

inisitijitty opened the pull request:
# Description

Updates mimir chart to version 3.3.*

# Issues

Fixes #868

# Changes

- add `kubeVersionOverride` value (default `null`)
- set `image.tag` to `2.4.0`
- set `enterprise.image.tag` to `2.4.0`
- remove `*.affinity` defaults values
- add `*.topologySpreadConstraints` block
- add `nginx.route` block (for OpenShift)
- add `metaMonitoring.grafanaAgent.logs.enabled` (default `true`)
- add `metaMonitoring.grafanaAgent.metrics.enabled` (default `true`)

# Checklist

* [x] This PR contains a description of the changes I'm making
* [x] I updated the version in Chart.yaml
* [x] I updated applicable README.md files using  `pre-commit run`
* [x] I documented any high-level concepts I'm introducing in `docs/`
* [ ] CI is currently green and this is ready for review
* [x] I am ready to test changes after they are applied and released

eyenx reviewed (commented):
Any big changes in mimir 2.4.0?

inisitijitty commented:
> Any big changes in mimir 2.4.0?

Nothing big from what I see.

# Grafana Mimir

    [CHANGE] Distributor: change the default value of -distributor.remote-timeout to 2s from 20s and -distributor.forwarding.request-timeout to 2s from 10s to improve distributor resource usage when ingesters crash. #2728 #2912
    [CHANGE] Anonymous usage statistics tracking: added the -ingester.ring.store value. #2981
    [CHANGE] Series metadata HELP that is longer than -validation.max-metadata-length is now truncated silently, instead of being dropped with a 400 status code. #2993
    [CHANGE] Ingester: changed default setting for -ingester.ring.readiness-check-ring-health from true to false. #2953
    [CHANGE] Anonymous usage statistics tracking has been enabled by default, to help Mimir maintainers make better decisions to support the open source community. #2939 #3034
    [CHANGE] Anonymous usage statistics tracking: added the minimum and maximum value of -ingester.out-of-order-time-window. #2940
    [CHANGE] The default hash ring heartbeat period for distributors, ingesters, rulers and compactors has been increased from 5s to 15s. Now the default heartbeat period for all Mimir hash rings is 15s. #3033
    [CHANGE] Reduce the default TSDB head compaction concurrency (-blocks-storage.tsdb.head-compaction-concurrency) from 5 to 1, in order to reduce CPU spikes. #3093
    [CHANGE] Ruler: the ruler's [remote evaluation mode](https://grafana.com/docs/mimir/latest/operators-guide/architecture/components/ruler/#remote) (-ruler.query-frontend.address) is now stable. #3109
    [CHANGE] Limits: removed the deprecated YAML configuration option active_series_custom_trackers_config. Please use active_series_custom_trackers instead. #3110
    [CHANGE] Ingester: removed the deprecated configuration option -ingester.ring.join-after. #3111
    [CHANGE] Querier: removed the deprecated configuration option -querier.shuffle-sharding-ingesters-lookback-period. The value of -querier.query-ingesters-within is now used internally for shuffle sharding lookback, while you can use -querier.shuffle-sharding-ingesters-enabled to enable or disable shuffle sharding on the read path. #3111
    [CHANGE] Memberlist: cluster label verification feature (-memberlist.cluster-label and -memberlist.cluster-label-verification-disabled) is now marked as stable. #3108
    [CHANGE] Distributor: only single per-tenant forwarding endpoint can be configured now. Support for per-rule endpoint has been removed. #3095

# Mixin

    [CHANGE] Alerts: MimirQuerierAutoscalerNotActive is now critical and fires after 1h instead of 15m. #2958
